### ISSUE TITLE
PhpSpreadsheet and Excel calculate `NULL <> 0` differently

### ISSUE THREAD

**Issue by ndench** [PHPOffice/PhpSpreadsheet#2523]
This is:

```
- [x] a bug report
- [ ] a feature request
- [ ] **not** a usage question
```

I've found an edge case where a calculation in Excel differs from the calculation in PhpSpreadsheet. It's for the formula `NULL<>0`, however, it's impossible to actually enter the value `NULL` into a cell though, so I use a cell reference that does not have data in it (eg. `B1<>0` where nothing exists in cell B1).

While I think that `NULL<>0` should definitely return `TRUE` from a logical sense, the fact the excel returns `FALSE` means that PhpSpreadsheet probably also should.

From a practical sense, this formula is often used when calculating costs based on `quantity * unit * rate`, where column `B` is the quantity, `D` is the unit price, and `E` is the rate (cost per unit). Sometimes the price of the item might be included in the overall price so it doesn't need a unit price itself, so we want to default the cost to be `"INCL.:

![Screenshot 2022-01-24 122708](https://user-images.githubusercontent.com/2062388/150712236-adeb4006-1361-4f52-abe5-f1b3b143474b.png)

In the above screenshot, the cells `E2:E5` hav the following formula, and cell `E3` gets the result of `B3<>0 === FALSE`.
```
=IF(B2<>0,B2*D2,"INCL.")
```

Here is the example worksheet if you're interested:
[example.xlsx](https://github.com/PHPOffice/PhpSpreadsheet/files/7922190/example.xlsx)

### What is the expected behavior?

In Excel, the formula `NULL<>0` returns `FALSE`. It's impossible to actually enter the value `NULL` into a cell though, so I use a cell reference that does not have data in it:

![Screenshot 2022-01-24 120436](https://user-images.githubusercontent.com/2062388/150710678-e7efe508-f3f0-4bf9-a6df-36a303b3b0ce.png)

### What is the current behavior?

Calculating this same value in PhpSpreadsheet results in `TRUE`.

### What are the steps to reproduce?

Here is the excel file I'm using. Created by opening a blank sheet and entering the value `=B1<>0` into cell A1.
[replication.xlsx](https://github.com/PHPOffice/PhpSpreadsheet/files/7922106/replication.xlsx)

When you open this file in excel, you'll notice the result of the formula is `FALSE`.

If you run the following example, you'll see that PhpSpreadsheet returns `true` for the same function.

```php
<?php

use PhpOffice\PhpSpreadsheet\Reader\Xlsx;

require __DIR__ . '/vendor/autoload.php';

$reader = new Xlsx();
$excel  = $reader->load('replication.xlsx');
$sheet = $excel->getSheet(0);

$cell = $sheet->getCell('A1');
$value = $cell->getCalculatedValue();

var_dump($value);
```

### Which versions of PhpSpreadsheet and PHP are affected?

I'm currently using the latest version (1.21.0).

**Comment by oleibman** [comment]
Although I can confirm that these errors happen with 1.21, it appears that they are fixed in master, probably due to PR #2537. If you can test against master, can you please confirm?

**Comment by oleibman** [comment]
The issue seems to have reappeared in 1.22. Investigating.
